runner/runner_test.go

```
@@ -768,3 +779,278 @@ func TestCreateNetworkpolicyInstance_AllowDenyFlags(t *testing.T) {
+	require.Len(t, results, 1)
+	require.Equal(t, "http", results[0].Scheme, "a cleartext 400 must stay http")
+	require.Equal(t, http.StatusBadRequest, results[0].StatusCode)
+}
+
+// TestOneShotPlainHTTPKeepsItsResult covers a cleartext service that answers
+// once and then refuses: the scheme decision must not cost it a second
+// request, or a reachable service disappears from the output.
+func TestOneShotPlainHTTPKeepsItsResult(t *testing.T) {
+	var served int64
+
+	listener, err := net.Listen("tcp", "127.0.0.1:0")
```

golangci-lint (2.13.2)

[error] net.Listen must not be called. use (*net.ListenConfig).Listen

(noctx)